cylc/flow/task_job_mgr.py

```diff
@@ -273,10 +273,10 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
             for itask in itasks:
                 # If there are any hosts left for this platform which we
                 # have not previously failed to contact with a 255 error.
-                if (
-                    set(itask.platform['hosts']) -
```

Comment:
This `set()` was removing duplicates, not sure if that was important.

There is an existing interface in cylc.flow.network.resolvers for producing unique lists if it helps.

https://github.com/cylc/cylc-flow/blob/7b4e95c7dfeac69100530d06b686c9345067fd58/cylc/flow/network/resolvers.py#L96-L111

Reply:
Probably not hugely, but here./..

```suggestion
                     list(set(itask.platform['hosts'])) - 
```

Reply:
I don't think so, because this is in essence user error, but I've put in a `list(set(...))` just to be safe.